src/main/java/pl/kruczala/michal/Casket100.java

```diff
@@ -0,0 +1,22 @@
+package pl.kruczala.michal;
+
+public class Casket100 extends Casket200 {
+    int numberOf100BanknotesInCasket = 200;
+    int usedNumberOf100BanknotesToPayOut = -1;
```

Comment:
jeśli gdzieś stosujesz -1 to jest znak dla Ciebie, że prawdopodobnie da się przekształcić warunki w taki sposób aby zacząć od 0. Generalnie jest to ok w świecie języka c/c++ ale wynika to raczej z różnego rodzaju ograniczeń i wydajności których w Javie zazwyczaj się nie przejmujesz